packages/mollitia/src/module/breaker/index.ts

```diff
@@ -183,26 +186,54 @@ export abstract class SlidingWindowBreaker<T> extends Module {
     this.reinitializeCounters();
   }
 
-  public async execute<T1> (circuit: Circuit, promise: CircuitFunction, ...params: any[]): Promise<T1> {
-    const _exec = this._promiseBreaker<T1>(circuit, promise, ...params);
+  public async execute<T>(circuit: Circuit, promise: CircuitFunction, ...params: any[]): Promise<T> {
+    let data;
+    try {
+      data = await this.getState();
+    } catch (e) {
+      console.warn('Cannot get state');
+    }
+    if (data?.requests) {
+      this.requests = (data.requests as SerializableRecord[]).map((r) => {
```

Comment:
I think casting for Serializable in this file might be dangerous, we might want to have typeguards to ensure the state retrieved from `redis` has the correct format.
We could use [zod](https://zod.dev/) that helps with that, or do that ourselves:

``` typescript
function isValidState (_state: unknown): _state is SlidingWindowBreakerState {
  // Do checks here and return a boolean
  // Is it's true, when we'll use this method, typescript will ensure the variable type is correct
  return false;
}
```